Protein-gym.py

```diff
@@ -50,33 +73,55 @@
 # %%
 # load substitution data, introduce whitespeces and CLS/EOS tokens
 # save complete data as csv, load as HF dataset
-if os.path.exists(path + "ProteinGym_substitutions.csv"):
+if os.path.exists(path + "ProteinGym_substitutions.csv") and os.path.exists(path + "ProteinGym_substitutions_base_seqs.csv"):
     print("preprocessing was already done, load csv")
     dataset = load_dataset("csv", data_files=(path + "ProteinGym_substitutions.csv"))
+    base_seqs = pd.read_csv(path + "ProteinGym_substitutions_base_seqs.csv")
 else:
     print("preprocess substitutions ...")
     folder_path = "data/ProteinGym/ProteinGym_substitutions"
     all_data = []
+    base_seqs = {} # init dict to save baseseqs
     for filename in os.listdir(folder_path):
         if filename.endswith('.csv'):
             file_path = os.path.join(folder_path, filename)
             df = pd.read_csv(file_path)
+            experiment = filename[:-4]
+            # save base_seqs, as experiment:sequence pair, needs to be before changing mutant name :)
+            base_seqs[experiment] = get_base_sequence(df)
+            # add experiment name to track and get base sequence for zero-shot tasks
+            df["mutant"] = experiment + "_" + df["mutant"]
             all_data.append(df)
+            
+    # get dataframe
     merged_data = pd.concat(all_data, ignore_index=True)
+    base_seqs = pd.DataFrame(base_seqs, index=["base_seq"]).T
+    base_seqs = base_seqs.reset_index(names="experiment")
+    # save the baseseqs
+    base_seqs.to_csv(path + "ProteinGym_substitutions_base_seqs.csv", index=False)
     # Add spaces between each amino acid in the 'mutated_sequences' column
-    merged_data['mutated_sequence'] = merged_data['mutated_sequence'].apply(lambda seq: ' '.join(list(seq)))
+    # merged_data['mutated_sequence'] = merged_data['mutated_sequence'].apply(lambda seq: ' '.join(list(seq)))
     # add cls and end tokens
-    merged_data['mutated_sequence'] = "<cls> " + merged_data['mutated_sequence'] + " <eos>"
+    merged_data['mutated_sequence'] = "<cls>" + merged_data['mutated_sequence'] + "<eos>"
     # save csv
     merged_data.to_csv(path + "ProteinGym_substitutions.csv", index=False)
     dataset = load_dataset("csv", data_files=(path + "ProteinGym_substitutions.csv"))
     del merged_data
 
 # %% tokenize, with esm2_t33_650M_UR50D, use same checkpoint for model
 checkpoint = "facebook/esm2_t33_650M_UR50D"
-tokenizer = AutoTokenizer.from_pretrained(checkpoint)
+# autoTokenizer = AutoTokenizer.from_pretrained(checkpoint)
+tokens = [
+            "<cls>", "<pad>", "<eos>", "L", "A", "G", "V", 
+            "S", "E", "R", "T", "I", "D", "P", "K", "Q", "N", 
+            "F", "Y", "M", "H", "W", "C", "B", "U", "Z", "O", 
+            "<mask>"
+        ]
+tokenizer = AptTokenizer(tokens)
```

Comment:
Once PR #14 get's merged, you don't have to define the tokens here, you can just do
```
tokenizer = AptTokenizer()
```